.travis.yml

```diff
@@ -27,78 +29,82 @@ DEPLOY_TO_GITHUB: &DEPLOY_TO_GITHUB
 
 jobs:
   include:
-    # OSX Binary
-    - env: JOB=dist-osx ARCH=x86_64-apple-darwin
-      os: osx
-      stage: build
+    - name: lint-checks
+      stage: test
+      addons:
+        apt:
+          packages: ['python3-pip']
+      install: pip3 install --user flake8==3.7.8
       script:
-        - cmake . && make
-      <<: *DEPLOY_TO_GITHUB
+        - flake8
+        - ./clang-tidy-diff.sh
+        - ./clang-format-diff.sh
+        # ensure generated parser is up to date
+        - ./scripts/gen-s-parser.py | diff src/gen-s-parser.inc -
 
     # Build with clang and run tests on the host system (Ubuntu).
     - &test-ubuntu
+      name: clang
       stage: test
       compiler: clang
-      python:
-        - 2.7
-        - 3.6
       addons:
         apt:
           sources: ['ubuntu-toolchain-r-test']
-          packages: ['cmake', 'g++-5', 'python3-pip']
+          packages: ['cmake', 'g++-5']
       before_install:
         - export ASAN_OPTIONS="$ASAN_OPTIONS symbolize=1"
       install:
         - nvm install 12
         - nvm use 12
-        - pip3 install --user flake8==3.7.8
         # get jsvu in order to get more js engines
         - npm install jsvu -g
         - export PATH="${HOME}/.jsvu:${PATH}"
         - jsvu --os=linux64 --engines=spidermonkey,v8
       script:
         - set -o errexit
-        - flake8
-        - ./clang-format-diff.sh
-        # ensure generated parser is up to date
-        - ./scripts/gen-s-parser.py | diff src/gen-s-parser.inc -
-        - BUILD_SUBDIR=${BUILD_SUBDIR:-.}
-        - mkdir -p ${BUILD_SUBDIR} && cd ${BUILD_SUBDIR}
-        - cmake ${TRAVIS_BUILD_DIR} -DCMAKE_C_FLAGS="$COMPILER_FLAGS" -DCMAKE_CXX_FLAGS="$COMPILER_FLAGS" -DCMAKE_EXE_LINKER_FLAGS="$LINKER_FLAGS" -DCMAKE_INSTALL_PREFIX=install -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
-        # clang-tidy-diff.sh may not exist when BUILD_SUBDIR is a subdirectory
-        - if [ -f clang-tidy-diff.sh ]; then ./clang-tidy-diff.sh; fi
```

Comment:
is this if and the comment explaining it no longer relevant? cc @aheejin

Reply:
Yeah.. the problem was the `cd` above which is no longer part of the lint phase. So is not long needed or relevant.  Much cleaner this way.